### PR TITLE
Register the magnet protocol

### PR DESCRIPTION
**Pull request by rllola**
From https://github.com/JoyStream/joystream-electron/issues/515

- [x] Added `protocolName` field (needed to register protocol)
- [x] Implement handlers for protocols (magnet, joystream) in main.js (Not using `joystream://` yet)
- [x] Actually load the torrent using the magnet

### CODE REVIEW

**Comment by rllola** [comment]
Adding a magnet seems to broke the app if the metadata has not been fetch and saved...

**Comment by mnaamani** [comment]
There seems to be an issue starting the app with torrents already in the database.

screen shot 2017-11-24 at 8 31 51 am

File: src/core/Application/Statemachine/Common.js [hunk 1]
```diff
@@ -66,7 +71,8 @@ function addTorrent(client, settings) {
     let params = {
         name: settings.name,
         savePath: settings.savePath,
-        ti: settings.metadata
+        ti: settings.metadata,
+        url: settings.url
     }
```
Comment (mnaamani) [inline src/core/Application/Statemachine/Common.js]: I think we should only have on key, either `ti` if it is not null or `url`.. not both.

Comment (rllola) [reply src/core/Application/Statemachine/Common.js]: If `url` is set `ti` is going to be ignored inside the libtorrent library : http://www.libtorrent.org/reference-Core.html#add_torrent_params

So having both should not be a problem.

Comment (mnaamani) [reply src/core/Application/Statemachine/Common.js]: OK, what about in joystream-node when decoding the params from javascript, does it handle null correctly?

Comment (mnaamani) [reply src/core/Application/Statemachine/Common.js]: Also for the case when a torrent is loaded from database url is `undefined` are we handling it correctly?

Comment (rllola) [reply src/core/Application/Statemachine/Common.js]: Yes it is handling it correctly inside the libtorrent-node binding library : https://github.com/JoyStream/libtorrent-node/blob/master/module/src/add_torrent_params.cpp

When stopping the application only torrent with metadata will be saved. And only the metadata torrent will be saved in the database so no mention of the magnet uri for this part.

**Comment by rllola** [comment]
I ma going to add more console log because I am not sure it is happening when loading the torrent from the database.

It is the error that I used to have when trying to add a magnet link that I already added... Might be also an error linked to the fact that it is already a fully download torrent ?

**Comment by rllola** [comment]
Pointing to the old repo.

**Pull request closed without merging**